diff --git a/profile2.admin.inc b/profile2.admin.inc
index 4394dc0..6fcbf04 100644
--- a/profile2.admin.inc
+++ b/profile2.admin.inc
@@ -64,6 +64,13 @@ function profile2_type_form($form, &$form_state, $profile_type, $op = 'edit') {
     '#value' => t('Save profile type'),
     '#weight' => 40,
   );
+  $form['weight'] = array(
+    '#type' => 'weight',
+    '#title' => t('Weight'),
+    '#default_value' => $profile_type->weight,
+    '#description' => t('When showing profiles, those with lighter (smaller) weights get listed before profiles with heavier (larger) weights.'),
+    '#weight' => 10,
+  );
 
   if (!$profile_type->isLocked() && $op != 'add' && $op != 'clone') {
     $form['actions']['delete'] = array(
diff --git a/profile2.module b/profile2.module
index 691e68d..e53317c 100644
--- a/profile2.module
+++ b/profile2.module
@@ -833,6 +833,32 @@ function profile2_field_access($op, $field, $entity_type, $profile = NULL, $acco
 }
 
 /**
+ * Implements hook_field_extra_fields().
+ *
+ * We need to add pseudo fields for profile types to allow for weight settings
+ * when viewing a user or filling in the profile types while registrating.
+ */
+function profile2_field_extra_fields() {
+  $extra = array();
+  foreach (profile2_get_types() as $type_name => $type) {
+    // appears on: admin/config/people/accounts/display
+    $extra['user']['user']['display']['profile_' . $type_name] = array(
+      'label' => t('Profile: @profile', array('@profile' => $type->label)),
+      'weight' => $type->weight,
+    );
+    // appears on: admin/config/people/accounts/fields
+    if (!empty($type->data['registration'])) {
+      $extra['user']['user']['form']['profile_' . $type_name] = array(
+        'label' => t('Profile: @profile', array('@profile' => $type->label)),
+        'description' => t('Appears during registration only.'),
+        'weight' => $type->weight,
+      );
+    }
+  }
+  return $extra;
+}
+
+/**
  * Entity metadata callback to load profiles for the given user account.
  */
 function profile2_user_get_properties($account, array $options, $name) {
